sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/BeamFnDataClient.java

```diff
@@ -61,4 +61,10 @@
       Endpoints.ApiServiceDescriptor apiServiceDescriptor,
       LogicalEndpoint outputLocation,
       Coder<WindowedValue<T>> coder);
+
+  /**
+   * Drain all element queues and cleanup all objects created for the handling of elements for the
+   * given apiServiceDescriptor representing the data channel and the instructionId.
+   */
+  void drainAndBlock(ApiServiceDescriptor apiServiceDescriptor, String instructionId);
```

Comment:
From the description and method name, its unclear when people using the BeamFnDataClient would ever use this method.

Also, from a code perspective, why would we need to change this?

The ProcessBundleHandler could create its own BeamFnDataClient that wraps the real BeamFnDataClient and it contains all the queue logic / receiver management. Since this would be internal to the ProcessBundleHandler, processing the queue would never be exposed.

Reply:
Made this change as we discussed offline